pre_processing/read_slstr_funcs.F90

```diff
@@ -250,7 +272,10 @@ subroutine read_slstr_visdata(indir, inband, outarr, imager_angles, &
    call ncdf_close(fid, 'read_slstr_visdata()')
 
    ! Resample the data to the TIR grid size.
-   call slstr_resample_vis_to_tir(data1, outarr, nx, ny, sreal_fill_value)
+   ! Ensure that the array passed to the resampling routine is the
+   ! expected size
+   call slstr_resample_vis_to_tir(data1(1:nx*2,1:ny*2), outarr, &
```

Comment:
I'm confused now. In theory, `read_ncdf_field` determines the size of the field to read from the array we give it for output. If the actual data is one bigger than what we expected, the read should go off without a hitch.

Could you send me a filename on JASMIN that crashes here, so I can work out what I did wrong?

Reply:
Ok, I've looked a bit further into this and it looks like the problem is that some frames, which appear to always be at the  have one extra row in both IR and VIS grids. Therefore, the vis channel has one _less_ row than the code is expecting, which would cause your code to fall over.
I agree with you, having gone through the whole ncdf reading nest of routines, that if there was an extra row in the visible and not the IR, it would read the data just fine.
I've attached a list of files which suffer from this problem (at least most of them do).
[missing_l2_slstrb_20190101_to_20191231-20230531.txt](https://github.com/ORAC-CC/orac/files/11663499/missing_l2_slstrb_20190101_to_20191231-20230531.txt)